### ISSUE TITLE
termsAndConditions should support optional selectboxes

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#26608]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

`termsAndConditions` hook allows to add other checkboxes to order confirmation. This hook however does not support adding optional checkboxes before order confirmation button. Modules need to solve it by forcing users to hardcode a custom hook into the template.

Majority of stores in Czech republic are using price comparing website called XY to list their product offers. This website also provides store rating & reviews and is automatically sending email with a request to review this store after a few days. The data is sent to their API after the order is created.

However, by law, the customer must be able to opt out from this reviews when ordering. So, a checkbox called with a text like `I don't agree with you sending me review request from XY after order.` must be available in checkout.

_______________

![Výstřižek](https://user-images.githubusercontent.com/6097524/141788833-3648536d-2eb3-4e31-a283-042ad77b25ea.JPG)

### Describe the solution you'd like

First solution:
- Add a possibility to add a checkbox with optional value to the current hook (https://github.com/PrestaShop/PrestaShop/blob/develop/classes/checkout/ConditionsToApproveFinder.php). The module will watch it's state via JS, as it is now.

Second solution, maybe more universal:
- Add a hook displaying any content before order confirmation button https://github.com/PrestaShop/PrestaShop/blob/develop/themes/classic/templates/checkout/_partials/steps/payment.tpl#L18

### Additional context

Ping @Matt75 - I think you could decide the best solution. ;-)

**Comment by florine2623** [comment]
Hello @Hlavtox ,

Thank you for your suggestion. The @PrestaShop/product-team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thank you

**Comment by Matt75** [comment]
Opt-out is a bad practice and is not recommended by GDPR.
This needs to be explored by @PrestaShop/product-team

> The GDPR requires consent to be opt-in. It defines consent as “freely given, specific, informed and unambiguous” given by a “clear affirmative action.” It is not acceptable to assign consent through the data subject’s silence or by supplying “pre-ticked boxes.”

Source : International Association of Privacy Professionals (IAPP) https://iapp.org/news/a/yes-how-opt-in-consent-really-works/

**Comment by Hlavtox** [comment]
@Matt75 Opt-in or opt-out, we can't do any of them. So, should we add `defaultState` and `optional` parameters to the TOS hook or just add a pure display hook and let people do whatever they want?

**Comment by Matt75** [comment]
> @Matt75 Opt-in or opt-out, we can't do any of them. So, should we add `defaultState` and `optional` parameters to the TOS hook or just add a pure display hook and let people do whatever they want?

That's true, currently every checkboxes needs to be checked in order to be allowed to pay, the hook is named `termsAndConditions` because it's the intended behavior, it was created to be compliant with regulation in EU. It was integrated into Core from [advancedeucompliance module for PrestaShop 1.6](https://github.com/PrestaShop/advancedeucompliance)

In some situation optional checkboxes can be legits but I thinks it should be managed in another hook named accordingly to avoid confusion.

It's better to wait @PrestaShop/product-team

**Comment by Hlavtox** [comment]
Still relevant

**Comment by boo-code** [comment]
The second option in @Hlavtox's question - "or just add a pure display hook and let people do whatever they want" - was implemented about two months after this thread, and nobody came back to say so.

`displayCheckoutBeforeConfirmation` was added on 2022-01-22 (`6c0340baccd` for the template, `0ac2ab11c57` for the install data). It is in `install-dev/data/xml/hook.xml` as "Show custom content before checkout confirmation", and both `classic-theme` and `hummingbird` render it in `templates/checkout/_partials/steps/payment.tpl`.

Placement is the part that matters here. The mandatory checkboxes live inside `<form id="conditions-to-approve">`, and the hook is rendered after that form closes, before the confirmation area. So a checkbox emitted by a module through this hook is not part of `conditions_to_approve` and is therefore not subject to the rule that every box must be ticked before paying - which is the blocker described at the top of this issue.

One consequence worth stating plainly: because it sits outside that form, and outside the payment module's own form, the value is not carried by the order submission on its own. A module rendering an input there wires the value itself. For the review opt-out case that is not a limitation, since the data is sent to the external API after the order exists anyway.

This does not settle the other half of the original request - `optional` and `defaultState` parameters on `termsAndConditions` itself - nor @Matt75's GDPR point, which the display hook does not decide either way: the module chooses whether its own box is opt-in or opt-out.
